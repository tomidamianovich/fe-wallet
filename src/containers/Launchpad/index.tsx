import React from 'react'
import MaintenancePage from '../../components/MaintenancePage'
import { withTranslation, WithTranslation } from 'react-i18next'
import './styles/index.scss'

type Props = WithTranslation

/*

  Component Name: Launchpad
  Usages: 
    - Launchpad Page
    
*/

const Launchpad: React.FC<Props> = ({ t }) => (
  <div className="fe-wallet__launchpad">
    <MaintenancePage pageName={t('pages.launchpad')} />
  </div>
)
export default withTranslation()(Launchpad)
