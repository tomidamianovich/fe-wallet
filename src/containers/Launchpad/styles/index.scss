.fe-wallet__launchpad {
  display: flex;
  width: 100%;
}
