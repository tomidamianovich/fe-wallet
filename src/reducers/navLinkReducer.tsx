import * as actionTypes from '../actionTypes/NavLinkType'
import { NavLinkAction, NavLinkState } from '../utils/type'
import { NAV_LINKS } from '../utils/constants'

const initialState: NavLinkState = NAV_LINKS

const reducer = (
  state: NavLinkState = initialState,
  action: NavLinkAction,
): NavLinkState => {
  switch (action.type) {
    case actionTypes.SET_NAV_LINKS:
      return action.payload
  }
  return state
}

export default reducer
