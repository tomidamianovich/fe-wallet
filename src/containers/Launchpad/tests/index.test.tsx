import { render } from '@testing-library/react';
import Launchpad from '../';
import navLinkReducer from "../../../reducers/navLinkReducer"
import { Provider } from 'react-redux';
import { createStore } from "redux";

const store = createStore(navLinkReducer);

test('renders Launchpad main page', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Launchpad />
    </Provider>);
  expect(asFragment()).toMatchSnapshot();
});