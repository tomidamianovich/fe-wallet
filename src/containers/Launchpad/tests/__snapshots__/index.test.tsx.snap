// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Launchpad main page 1`] = `
<DocumentFragment>
  <div
    class="fe-wallet__launchpad"
  >
    <div
      class="fe-wallet__maintenance-page"
    >
      <h1>
        Pagina en Mantenimiento
      </h1>
      <h3>
        Oops! Estamos trabajando en la pagina . Ingrese nuevamente a esta seccion m√°s tarde.
      </h3>
    </div>
  </div>
</DocumentFragment>
`;
